changelog/_unreleased/2023-02-02-add-admin-script.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Adds admin scripts - Create administration pages with twig
+issue: NEXT-
+author: Heiner Lohaus
+author_email: heiner@lohaus.eu
+author_github: hlohaus
+---
+# Admin scripts
+* Added admin script controller for own app pages in the administartion.
+
+## Security
+* New security service with nonce function:
+```
+<script nonce="{{ services.security.nonce() }}"></script>
+```
+* Create response with disabled content security:
+```
+{{ set response = services.security.response(content, '') }}
+```
+* Iframes with same origin are supported in admin scripts. 
+```
+X-Frame-Options: sameorigin
+```
+## Admin-Router
+* Create simply admin urls with the router service:
+```
+{{ services.router.generate('admin-hook', query) }}
+```
+* Or redirect the request to a other admin script:
+```
+{% do services.router.redirect('admin-hook', query) %}
+```
+
+## Bug fix
+* Bug fix: Internal resolved actions buttons now includes the complete payload. `Shopware\Core\Framework\App\ActionButton\Executor`
+* Bug fix: Use translations in your scripts. Example: 
+```
+{{ 'general.description'|trans({}, null, 'en-GB') }}
```

Comment:
Nice, but i think providing a hard coded locale is a edge case, by default the correct language based on the current context should be used

Reply:
Currently i use:  {% set lang = hook.query['sw-user-language'] %}
Maybe we can fix it too.

Reply:
Should it add a setScriptsLocale method to the translator? @keulinho 
And automatic set it in a listener.

Reply:
I added a listener.